src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1798,6 +1932,38 @@ def nic_attach_choices(self):
         """
         return self.middleware.call_sync('interface.choices', {'exclude': ['epair', 'tap', 'vnet']})
 
+    @accepts()
+    def pptdev_choices(self):
+        """
+        Available choices for PCI passthru device.
+        """
+        if not self.pptdevs:
+            proc = None
+            try:
+                proc = subprocess.Popen(['/usr/sbin/pciconf', '-l'],
```

Comment:
Same here please, we should be fine with just `run`

Reply:
@sonicaj, I am not _aware_ of a _current_ issue with `acpidump` or `sysctl`, apart from potentially generating a lot of output depending on how they are invoked. From a maintenance, robustness and security point of view, however, it may be beneficial to consider not only the current situation, but also how to detect and mitigate it in case the situation changes or there is an error. E.g., due to a bug triggered only under certain circumstances or due to an upgrade of the external tools otherwise unrelated to middlewared. I think it would be a sound principle to apply more generally and personally I would favour an evolution of the FN/TN async framework to provide this protection. 

As this is not my project and I understand you have more things to consider, e.g., code maintenance, I am fine with changing to `run`. Hoping to see a future evolution of the framework though to offer isolation/protection from unexpected behaviour/misbehaviour of external utilities.

Reply:
@sonicaj, the PR is updated according to our discussion so far and ready for review.

Reply:
Thank you for your concerns @stattin42, if we notice such a case ( or as time permits ), we will implement adequate measures to ensure we are good.